src/charm.py

```diff
@@ -714,22 +750,39 @@ def _is_bessd_grpc_service_ready(self) -> bool:
     def _is_bessd_configured(self) -> bool:
         """Check if bessd has been configured.
 
-        Examine the output from bessctl to show worker. If there is no
-        active worker, bessd is assumed not to be configured.
+        Bess is considered as configured when the `accessRoutes` and `coreRoutes` modules
+        are available and at least one worker is in the RUNNING state. This method examines
+        the output of the `bessctl show` command to determine whether the above criteria are met.
 
         Returns:
-            bool:   True/False
+            bool: Whether the `bessd` service is configured
         """
-        command = "/opt/bess/bessctl/bessctl show worker"
+        show_accessRoutes_module_cmd = "/opt/bess/bessctl/bessctl show module accessRoutes"  # noqa: N806
+        show_coreRoutes_module_cmd = "/opt/bess/bessctl/bessctl show module coreRoutes"  # noqa: N806
+        show_worker_cmd = "/opt/bess/bessctl/bessctl show worker"
         try:
-            (stdout, _) = self._exec_command_in_bessd_workload(
-                command=command,
+            (show_accessRoutes_module_stdout, _) = self._exec_command_in_bessd_workload(  # noqa: N806
+                command=show_accessRoutes_module_cmd,
+                timeout=10,
+            )
+            logger.debug(
+                "bessd configured accessRoutes module: %s", show_accessRoutes_module_stdout
+            )
+            (show_coreRoutes_module_stdout, _) = self._exec_command_in_bessd_workload(  # noqa: N806
+                command=show_coreRoutes_module_cmd,
+                timeout=10,
+            )
+            logger.debug("bessd configured coreRoutes module: %s", show_coreRoutes_module_stdout)
+            (show_worker_stdout, _) = self._exec_command_in_bessd_workload(
+                command=show_worker_cmd,
```

Comment:
I think this could be the first check. If there are no workers, there should not be any modules.

Reply:
Done.